csharp_dotnetcore/51.Enterprise-Bot-Template/EnterpriseBot.cs

```diff
@@ -14,29 +15,27 @@ namespace EnterpriseBot
     /// </summary>
     public class EnterpriseBot : IBot
     {
-        private readonly BotServices _services;
-        private readonly SemaphoreSlim _semaphore;
-        private readonly EnterpriseBotAccessors _accessors;
+        private BotServices _services;
+        private ConversationState _conversationState;
+        private UserState _userState;
+        private DialogSet _dialogs;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="EnterpriseBot"/> class.
         /// </summary>
         /// <param name="botServices">Bot services.</param>
-        /// <param name="accessors">Bot State Accessors.</param>
-        public EnterpriseBot(BotServices botServices, EnterpriseBotAccessors accessors)
+        /// <param name="conversationState">Bot conversation state.</param>
+        /// <param name="userState">Bot user state.</param>
+        public EnterpriseBot(BotServices botServices, ConversationState conversationState, UserState userState)
         {
-            _accessors = accessors;
+            _conversationState = conversationState;
+            _userState = userState;
```

Comment:
check + throw on null for all 3 parameters
_conversationState = conversationState ?? throw new ArgumentNullException(nameof(conversationState));